src/app/modules/bounce/bounce.service.ts

```diff
@@ -176,7 +224,9 @@ export const updateBounces = async (body: ISnsNotification): Promise<void> => {
   if (oldBounces) {
     oldBounces.merge(latestBounces, notification)
   }
-  const bounce = oldBounces ?? latestBounces
-  logCriticalBounce(bounce, submissionId, bounceInfo)
-  await bounce.save()
+  const bounceDoc = oldBounces ?? latestBounces
+  if (bounceDoc.isCriticalBounce()) {
+    await handleCriticalBounce(bounceDoc, submissionId, bounceInfo)
+  }
```

Comment:
this function is doing too much by performing two commands - updating the bounce doc and also handling the critical bounce. instead, i'd suggest refactoring to returning the bounce doc to the controller instead, so that it can retain _control_ and decide how to use the bounce domain object.